File: warmUp/countValleys.py
testString = "UDDDUDUU"
num = len(testString)


def countingValleys(n,s):
    sea_level = 0
    valley_count = 0
    splitString = list(s)
    for i in range(0,n):
        if splitString[i] == "U":
            sea_level += 1
            if sea_level == 0:
                valley_count += 1
        elif splitString[i] == "D":
            sea_level -= 1
    print (valley_count)


countingValleys(num, testString)

File: warmUp/countValleys.js
let path = "DUDDDUUDUU";
let pathArray = path.split("")
let numberOfSteps = pathArray.length;

function countingValleys(num, string) {
    let pathArray = string.split("")
    let numberOfSteps = pathArray.length;
    while (numberOfSteps > 0) {
        let valleyCount = 0; 
        let seaLevel = 0; 
        for(let i = 0; i < pathArray.length; i++){
            if(pathArray[i] == "D"){
                seaLevel--;
                numberOfSteps--;
            } else if (pathArray[i] == "U"){
                seaLevel++;
                numberOfSteps--;
                if(seaLevel == 0){
                    valleyCount++;
                }
            }
        }
        return(valleyCount);
    }
}

countingValleys(numberOfSteps, path);